pkg/reloader/reloader.go

```diff
@@ -243,7 +271,11 @@ func (r *Reloader) Watch(ctx context.Context) error {
 		if err := r.apply(applyCtx); err != nil {
 			r.configApplyErrors.Inc()
 			level.Error(r.logger).Log("msg", "apply error", "err", err)
+			r.lastConfigApplySuccess.Set(0)
+			continue
 		}
+		r.lastConfigApplySuccess.Set(1)
+		r.lastConfigApplySuccessTimestamp.SetToCurrentTime()
```

Comment:
Do we need to have two metrics for `config apply` and `reload` here? As I can see, if the reload fails, then the config apply fails as well. WDYT?

Reply:
I saw we were maintaining counters for both so just followed the format. But happy to remove the `apply` metrics if they add little value?

Reply:
I think we can have both if we want - or add label for all metrics like `op=reload` or `op=apply`. But I would argue that if you know reload fail, you can dive later on why

Reply:
No need to have root cause metric here

Reply:
Removed these, on reflection I don't think they are important from an alerting persepctive.